assemblies/pentaho-solutions/src/main/resources/pentaho-solutions/system/pentahoServices.spring.xml

```diff
@@ -132,6 +132,7 @@
   <bean class="org.pentaho.platform.web.http.api.resources.FileResource" scope="request">
     <property name="whitelist"><ref bean="RepositoryDownloadWhitelist"/></property>
   </bean>
+  <bean class="org.pentaho.platform.web.http.api.resources.FileResourceGeneratedContentForSchedule" scope="request"/>
```

Comment:
This change is taking place on a file that'll be eligible for "upgrade path for 9.0" ( i.e. migrating conf. files from a previous version to 9.0 ).

- If a `pentahoServices.spring.xml` from previous pentaho-server version gets ported over, will this separation have a graceful fallback behaviour ?
- Should this PR be accompanied of updated to the migration scripts, to inject this line in a ported `pentahoServices.spring.xml` ?
- Should this have to be added to the upgrade path documentation ?

Reply:
No, it will not fail gracefully. The server will boot up with no errors, but the endpoint `/repo/files/generatedContentForSchedule` will not be available.

The separation was done so that we can use `FileResource` services in the "slim" CDE assembly, where scheduling and related libs were removed. If this change can't be performed automatically by the upgrade script, I guess it is odd to explain to clients why the change is needed... Your call.

Reply:
That endpoint was introduced in https://github.com/pentaho/pentaho-platform/pull/1970 for no apparent reason. I've found no usages of it, so we decided to remove it.